src/main/java/play/dev/filewatch/FileWatchService.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.dev.filewatch;
+
+import java.io.File;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+/** A service that can watch files */
+public interface FileWatchService {
+
+  /**
+   * Watch the given sequence of files or directories.
+   *
+   * @param filesToWatch The files to watch.
+   * @param onChange A callback that is executed whenever something changes.
+   * @return A watcher
+   */
+  FileWatcher watch(Iterable<File> filesToWatch, Runnable onChange);
+
+  /** @deprecated Use {@link #watch(Iterable, Runnable)} instead */
+  @Deprecated(since = "3.0.0", forRemoval = true)
+  default FileWatcher watch(List<File> filesToWatch, Callable<Void> onChange) {
+    return watch(filesToWatch, () -> onChange.call());
+  }
+
+  /**
+   * @deprecated Renamed to {@link #detect(int, LoggerProxy)}, the first argument targetDirectory of
```

Comment:
I kept this methods, so it does not break existing Play users. Even if `LoggerProxy` is not binary compatible, it is source compatible, and calling it from sbt will work.